### PR TITLE
Wedding endpoints km

### LINKED ISSUE #53: User should be able to add a new wedding (vendor dashboard)

### LINKED ISSUE #55: Refactor from fake weddings data to API weddings data

### PR DESCRIPTION
**Pull request by Kristiannmiller**
# Pull Request

## What's this PR do? 

- [X] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Testing
- [ ] Styling
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Description of changes?
- Adds api calls for all weddings and add wedding
- Retrieve all weddings from local server on VenderDashboard
- Add a wedding through the AddWeddingForm component

## Any background context you want to provide? 
Still waiting on a few more endpoints for single wedding and guest/photo list information

## What are the relevant tickets? 
* Closes #53 
* Closes #55

* Addresses #11

### CODE REVIEW

File: src/apiCalls.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+export const getWeddings = () => {
+  return fetch('http://localhost:8000/api/v1/weddings/weddings/')
+  .then(response => {
+    if(response.ok) {
+      return response.json()
+    } else {
+      throw Error
+    }
```
Comment (ericberglund117) [inline src/apiCalls.tsx:8]: Super pumped to finally get a successful API call working

File: src/apiCalls.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  })
+  .then(response => {
+    if(response.ok) {
+      return response
+    } else {
+      throw Error
+    }
+  })
+  .catch(error => {
+    alert('Could not add wedding')
```
Comment (ericberglund117) [inline src/apiCalls.tsx:31]: Do we need to isolate these more for testing down the road?

File: src/components/AddWeddingForm/AddWeddingForm.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+          onChange={event => setImage(event.target.value)}
+        />
+        <StyledButton onClick={event => submitWedding(event)}>
+          <div id="translate"></div>
+            <a className="link" id="addListButton">Submit Wedding</a>
+        </StyledButton>
+      </form>
+    </>
+	)
+}
```
Comment (ericberglund117) [inline src/components/AddWeddingForm/AddWeddingForm.tsx:89]: Looks great. Glad the wedding cards pop up automatically through the useEffect

**Review by ericberglund117 — APPROVED** [review]

File: src/apiCalls.tsx (hunk 1 above)
Comment (hannakim91) [reply src/apiCalls.tsx:8]: Thank you @IamNorma @fieldstyler !

File: src/apiCalls.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(wedding)
+  })
+  .then(response => {
+    if(response.ok) {
+      return response
+    } else {
+      throw Error
```
Comment (hannakim91) [inline src/apiCalls.tsx:27]: Should make some issues for user friendly error handling if the data can't be retrieved

File: src/apiCalls.tsx (hunk 2 above)
Comment (hannakim91) [reply src/apiCalls.tsx:31]: Could you explain what you mean?

File: src/components/AddWeddingForm/AddWeddingForm.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import './AddWeddingForm.css'
+import empty from '../../assets/emptyGuestList.png'
+import { postAWedding } from '../../apiCalls'
+import { StyledButton } from '../App/styledComponents.styles'
+
+
+type NewWedding = {
```
Comment (hannakim91) [inline src/components/AddWeddingForm/AddWeddingForm.tsx:8]: Refactor opportunity: export this to the dashboard so we don't set type for Wedding twice

File: src/components/AddWeddingForm/AddWeddingForm.tsx (hunk 3 above)
Comment (hannakim91) [reply src/components/AddWeddingForm/AddWeddingForm.tsx:89]: This page is almost complete!!

**Review by hannakim91 — APPROVED** [review]

**Pull request merged**